.github/workflows/ci.yml

```diff
@@ -139,9 +142,43 @@ jobs:
           name: spectacular-package
           path: dist/packages/spectacular
           if-no-files-found: warn
+      - name: 'Spectacular: Upload docs build artifact'
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.spectacular-docs-package-name }}
+          path: dist/packages/spectacular-docs
+          if-no-files-found: warn
+
+  deploy-docs:
+    name: Deploy docs to GitHub Pages
+    # if: >
+    #   github.ref == 'refs/heads/main' && (github.event_name ==
+    #   'workflow_dispatch' || github.event_name == 'push')
+    needs: build
+    runs-on: ubuntu-latest
+
+    env:
+      package-path: ./spectacular-docs/
+
+    steps:
+      # necessary for JamesIves/github-pages-deploy-action
+      - uses: actions/checkout@v2
+
+      - run: echo ${{ github.ref }}
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: ${{ env.spectacular-docs-package-name }}
+          path: ${{ env.package-path }}
+
+      - name: Deploy docs to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@4.1.3
+        with:
+          branch: gh-pages
```

Comment:
How is the lifecycle of this branch?

Reply:
The Docusaurus website is pushed to the `gh-pages` branch by this action. GitHub Pages then triggers a deployment.